Feature/copy url

noahfx opened the pull request:
Implementation of hotkeys to copy submission permalink and submission url, this resolves issue https://github.com/michael-lazar/rtv/issues/346

michael-lazar commented:
Thanks, I really appreciate what you did with the clipboard.py file! 

Comments:

- pyperclip still needs to be removed from setup.py
- unused ``sys`` import
- we need to add the new key bindings to the ``CONTROLS.rst`` document as well.
- Doesn't work for selfpost links (e.g. it copies the display name "self.explainlikeimfive" instead of the actual URL).  This should be an easy fix by using ``data['url_full']`` instead of ``data['url']``.
- You should use ``six.text_type()`` instead of ``str()`` to be compatible with python 2, otherwise ``.format()`` is mixing unicode with bytes which I'm pretty sure raises an exception. Actually, I think you can get away with sending ``.format(e)`` the raw error object and it will call ``__unicode__`` implicitly.
- I think it would be useful to flash a "Copied to clipboard" notification on success to give some user feedback that it worked, maybe with a short timeout of ~1 second.
- Needs some unit tests. I understand that it's difficult to setup the test environment for this project, so if necessary this is something that I can do myself before publishing the next release. For example I didn't test it, but theoretically if the subprocess raised an OSError it would crash the entire app.

noahfx commented:
I have changed all you asked and added a test
Let me know if the test is ok, by now it is passing in my local enviroment not sure if I wrote it in the right test file  `tests/test_submission.py`

michael-lazar commented:
I made a couple of minor changes and went ahead and merged your branch. Let me know if you have any questions.  I tried it out on my mac and found that ``os.name == 'mac'`` didn't actually work (it still returns ``posix``), so I switched to using ``sys.platform`` because it was already being used elsewhere in the codebase. The other big thing that I did was split your test into 2 separate test cases, one for OS X and one for Linux. I also altered the test so that it could still pass without 'xsel' or 'xcopy' installed, which is why it was failing on Travis. Thanks for the PR!
